Modules/FindESMF.cmake

```diff
@@ -72,59 +71,53 @@ if (ESMF_FOUND)
     endif()
   endforeach()
 
-  # Construct ESMF_VERSION from ESMF_VERSION_STRING_GIT
-  if(ESMF_FOUND)
```

Comment:
You should not remove the version checking. Using `ESMF_VERSION_STRING` is not robust because it will include `beta snapshot`. Using `ESMF_VERSION_MAJOR`, etc is better.